cpp/tests/test_sde_sir.cpp

```diff
@@ -0,0 +1,166 @@
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include "abm_helpers.h"
+#include "memilio/math/euler.h"
+#include "sde_sir/model.h"
+#include "sde_sir/simulation.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+const mio::ssir::Model& ssir_testing_model()
+{
+    static mio::ssir::Model model;
+    model.step_size = 0.25;
+    model.populations.array().setConstant(1);
+    { // set parameters s.t. coeffStoI is 1
+        model.parameters.set<mio::ssir::TimeInfected>(1);
+        model.parameters.set<mio::ssir::TransmissionProbabilityOnContact>(3);
+        model.parameters.get<mio::ssir::ContactPatterns>().get_baseline()(0, 0) = 1;
+    }
+    return model;
+}
+
+TEST(TestSdeSir, get_flows)
+{
+    // make two get_flows computations with mocked rng
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(4))
+        .WillOnce(testing::Return(1.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(0.))
+        .WillOnce(testing::Return(1.));
+
+    const Eigen::Vector3d y = Eigen::Vector3d::Constant(1);
+    Eigen::Vector2d flows   = Eigen::Vector2d::Constant(1);
+
+    // results contain two parts : deterministic + stochastic
+
+    ssir_testing_model().get_flows(y, y, 0, flows);
+    auto expected_result = Eigen::Vector2d{1 + 2, 1};
+    EXPECT_EQ(flows, expected_result);
+
+    ssir_testing_model().get_flows(y, y, 0, flows);
+    expected_result = Eigen::Vector2d{1, 1 + 2};
+    EXPECT_EQ(flows, expected_result);
+}
+
+TEST(TestSdeSir, Simulation)
+{
+    // make a single integration step via a simulation
+    // this should overwrite the model step size
+    ScopedMockDistribution<
+        testing::StrictMock<MockDistribution<mio::DistributionAdapter<std::normal_distribution<double>>>>>
+        normal_dist_mock;
+
+    EXPECT_CALL(normal_dist_mock.get_mock(), invoke)
+        .Times(testing::Exactly(2))
+        .WillOnce(testing::Return(.5))
+        .WillOnce(testing::Return(.5));
+
+    auto integrator = std::make_shared<mio::EulerIntegratorCore>();
+    auto sim        = mio::ssir::Simulation(ssir_testing_model(), 0, 1);
+    sim.set_integrator(integrator);
+    sim.advance(1);
+
+    EXPECT_EQ(sim.get_model().step_size, 1.0); // set by simulation
```

Comment:
Don't understand why this is 1.0. Isn't the default 0.1? In the `model` and `Simulation`?